compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,251 @@
+		},
+		"a zero": {
+			a:    0 * time.Second,
+			b:    time.Second,
+			want: 0,
+		},
+		"same unit - a smaller": {
+			a:    2 * time.Second,
+			b:    10 * time.Second,
+			want: 2 * time.Second,
+		},
+		"same unit - b smaller": {
+			a:    10 * time.Second,
+			b:    2 * time.Second,
+			want: 2 * time.Second,
+		},
+		"different unit - a smaller": {
+			a:    2 * time.Second,
+			b:    7 * time.Hour,
+			want: 2 * time.Second,
+		},
+		"different unit - b smaller": {
+			a:    2 * time.Second,
+			b:    100 * time.Millisecond,
+			want: 100 * time.Millisecond,
+		},
+	}
+	for k, v := range cases {
+		s.Run(k, func() {
+			actual := min(v.a, v.b)
+			s.Equal(v.want, actual)
+		})
+	}
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoffTable() {
+	cases := map[string]struct {
+		initial         time.Duration
+		maxBackoff      time.Duration
+		expectedBackoff time.Duration
+	}{
+		"next backoff should not hit the limit of 30s": {
+			initial:         10 * time.Second,
+			maxBackoff:      30 * time.Second,
+			expectedBackoff: 15 * time.Second,
+		},
+		"next backoff should not be higher than the limit of 10s": {
+			initial:         8 * time.Second,
+			maxBackoff:      10 * time.Second,
+			expectedBackoff: 10 * time.Second,
+		},
+	}
+	for name, c := range cases {
+		s.Run(name, func() {
+			cs := NewCachingScanner(mockScanner{}, "", c.initial, c.initial, c.maxBackoff, func(time.Duration) {})
+			s.Equal(c.expectedBackoff, cs.calcNextBackoff(c.initial))
+		})
+	}
+}
+
+func (s *TestComplianceCachingSuite) TestReadInventoryWrapFaultyUnmarshal() {
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	brokenWrap := "{\"CachedInventory\":42}"
+	err := os.WriteFile(inventoryCachePath, []byte(brokenWrap), 0600)
+	s.NoError(err)
+
+	actual := readInventoryWrap(inventoryCachePath)
+
+	s.Nil(actual) // We expect nil, but no panic even when the unmarshal fails
+}
+
+func (s *TestComplianceCachingSuite) TestScanReadFaultyCachedInventory() {
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	w := &inventoryWrap{
+		CacheValidUntil:      time.Now().Add(2 * time.Minute),
+		RetryBackoffDuration: 0,
+		CachedInventory:      "{\n  \"nodeId\": \"notvalid\", \"LANGUAGE_CVES_UNAVAILABLE\"\n  ]\n}",
+	}
+	s.writeWrap(w, inventoryCachePath)
+
+	actualInventory, actualValidity := readCachedInventory(inventoryCachePath)
+
+	// both actual values should be empty
+	s.Nil(actualInventory)
+	s.Equal(time.Time{}, actualValidity)
+}
+
+func (s *TestComplianceCachingSuite) TestScanWithoutExistingCacheWritesCache() {
+	initial := 8 * time.Second
+	cache := initial
+	maxBackoff := 10 * time.Second
+	inventoryCachePath := fmt.Sprintf("%s/inventory-cache", s.T().TempDir())
+	cs := *NewCachingScanner(mockScanner{}, inventoryCachePath, cache, initial, maxBackoff, func(time.Duration) {})
+	nodeName := "testme"
+
+	actual, err := cs.Scan(nodeName)
+	ci := s.readWrapToInventory(inventoryCachePath)
+	s.NoError(err)
```

Comment:
I suggest reordering these lines for better clarity:
```suggestion
	actual, err := cs.Scan(nodeName)
	s.NoError(err)

	ci := s.readWrapToInventory(inventoryCachePath)
```